### PR TITLE
Don't have negative timeouts

### PR DESCRIPTION
**Pull request by achamayou**

### CODE REVIEW

File: tests/infra/clients.py [hunk 1]
```diff
@@ -449,6 +448,7 @@ def request(self, request):
             except requests.exceptions.ReadTimeout:
                 raise TimeoutError
             time.sleep(0.1)
+            self.connection_timeout -= 0.1
```
Comment (eddyashton) [inline tests/infra/clients.py]: I've been trying a few variations of this within my PR as well. We should do the same thing in the CurlClient, and ideally have this timeout for `signed_request` as well as `request`.

Comment (eddyashton) [reply tests/infra/clients.py]: Also I think we should decrement while handling `SSLError`, before we check the timeout. This way round, when we actually go negative we'll return `None` from this function, but I think we want to `raise CCFConnectionException`.

Comment (achamayou) [reply tests/infra/clients.py]: How about:

```
self.connection_timeout -= 0.1
if self.connection_timeout < 0:
  self.connection_timeout = 0
```

and then the connection timeout check on :446 become equality?

Comment (eddyashton) [reply tests/infra/clients.py]: That still means we'll return `None`, as on the next loop the `while` will fail?

Comment (achamayou) [reply tests/infra/clients.py]: We can raise if we fall through?

**Comment by achamayou** [comment]
Doing a pass on clients.py to remove the Stream() sequencer, so I will fix this properly in that PR.

**Pull request closed without merging**